src/transformers/utils/quantization_config.py

```diff
@@ -271,3 +289,90 @@ def to_diff_dict(self) -> Dict[str, Any]:
+    """
+
+    quant_method: QuantizationMethod = QuantizationMethod.GPTQ
+    bits: int = field(
+        default=None, metadata={"help": "The number of bits to quantize to, supported numbers are (2, 3, 4, 8)."}
+    )
+    dataset: Union[List[str], str] = field(
+        default=None,
+        metadata={
+            "help": "The dataset used for quantization. You can provide your own dataset in a list of string"
+            "or just use the original datasets used in GPTQ paper ['wikitext2','c4','c4-new','ptb','ptb-new']"
+        },
+    )
+    group_size: int = field(
+        default=128,
+        metadata={
+            "help": "The group size to use for quantization. Recommended value is 128 and -1 uses per-column quantization."
+        },
+    )
+    damp_percent: float = field(
+        default=0.01,
+        metadata={
+            "help": "The percent of the average Hessian diagonal to use for dampening. Recommended value is 0.01."
+        },
+    )
+    dataset: Union[str, List[str]] = field(
+        default=None,
+        metadata={
+            "help": "The percent of the average Hessian diagonal to use for dampening. Recommended value is 0.01."
+        },
+    )
+    desc_act: bool = field(
+        default=True,
+        metadata={
+            "help": "Whether to quantize columns in order of decreasing activation size."
+            "Setting it to False can significantly speed up inference but the perplexity may become slightly worse."
+            "Also known as act-order."
+        },
+    )
+    sym: bool = field(default=True, metadata={"help": " Whether to use symetric quantization."})
+    true_sequential: bool = field(
+        default=True,
+        metadata={
+            "help": "Whether to perform sequential quantization even within a single Transformer block."
+            "Instead of quantizing the entire block at once, we perform layer-wise quantization."
+            "As a result, each layer undergoes quantization using inputs that have passed through the previously quantized layers."
+        },
+    )
+    pack_sequentially: bool = field(
+        default=False,
+        metadata={
+            "help": "Whether to pack the layer just after it is quantized. If False, we will pack the model at the end."
+        },
+    )
+    use_cuda_fp16: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to use optimized cuda kernel for fp16 model. Need to have model in fp16."},
+    )
+    model_seqlen: int = field(default=None, metadata={"help": "The maximum sequence length that the model can take."})
+    block_name_to_quantize: str = field(default=None, metadata={"help": "The transformers block name to quantize."})
+    module_name_preceding_first_block: List[str] = field(
+        default=None, metadata={"help": "The layers that are preceding the first Transformer block."}
+    )
+
+    batch_size: int = field(default=1, metadata={"help": " The batch size of the dataset"})
+
```

Comment:
```suggestion
    model_name_or_path: str = field(
        default=None, metadata={"help": "Used for backward compatibility"}
    )

    model_file_base_name: str = field(
        default=None, metadata={"help": "Used for backward compatibility"}
    )
```
I had to add these fields to make it work for the checkpoints of TheBloke

Reply:
I will change the structure of the dataclass to have something similar to BnbQuantizationConfig. This way it should ignore the kwargs that are not present and we don't need to add these fields.